src/core/config/column_index/option.cpp

```diff
@@ -0,0 +1,31 @@
+#include "config/column_index/option.h"
+
+#include "config/column_index/validate_index.h"
+
+namespace config {
+
+ColumnIndexOption::ColumnIndexOption(
+        std::string_view name, std::string_view description,
+        typename Option<config::IndexType>::DefaultFunc get_default_value)
```

Comment:
what's the point of `typename` here? It's already a fully resolved type due to `<config::IndexType>`